### PR TITLE
Prac 5 review request

### PR DESCRIPTION
**Pull request by CharlotteCorbett**
Please check for variable names, incorrect formatting, and anything else that can be done better.
@j-finger

### CODE REVIEW

File: Prac 5/state_names.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+"""
+CP1404/CP5632 Practical
+State names in a dictionary
+File needs reformatting
+"""
+
+states_dict = {"QLD": "Queensland", "NSW": "New South Wales", "NT": "Northern Territory", "WA": "Western Australia",
+               "ACT": "Australian Capital Territory", "VIC": "Victoria", "TAS": "Tasmania"}
```
Comment (j-finger) [inline Prac 5/state_names.py:8]: Does states_dict change? Could it be name slightly DIFFERENTLY?

File: Prac 5/hex_colours.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+print(COLOURS_DICT)
+user_choice = input("Colour:")
+while user_choice != '':
+    try:
+        print("{} - {}".format(user_choice, COLOURS_DICT[user_choice]))
+        user_choice = input("Colour:")
+    except KeyError:
+        print("{} does not exist".format(user_choice))
+        user_choice = input("Colour:")
```
Comment (j-finger) [inline Prac 5/hex_colours.py]: Does this line need to be duplicated (line 10 & 13)? Could it be placed somewhere once else for the same effect? (think indentation)

File: Prac 5/hex_colours.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
+COLOURS_DICT = {'antiquewhite1': '#ffefdb', 'antiquewhite2': '#eedfcc', 'antiquewhite3': 'cdc0b0',
+                'antiquewhite4': '#8b8378', 'aquamarine1': '#7fffd4', 'aquamarine2': '#76eec6',
+                'aquamarine4': '#458b74', 'azure1': '#f0ffff', 'azure2': '#e0eeee', 'azure3': '#c1cdcd'}
+
+print(COLOURS_DICT)
+user_choice = input("Colour:")
```
Comment (j-finger) [inline Prac 5/hex_colours.py]: user_choice is a little ambiguous,  what is the user choosing?

File: Prac 5/state_names.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+# while state_code != "":
+#     if state_code in states_dict:
+#         print(state_code, "is", states_dict[state_code])
+#     else:
+#         print("Invalid short state")
+#     state_code = input("Enter short state: ").upper()
+# for loop
+for state in states_dict:
+    state_name = states_dict[state]
+    print("{} is {}".format(state, state_name))
```
Comment (j-finger) [inline Prac 5/state_names.py:21]: Is this what the question is asking for? Does the user have any input?

File: Prac 5/word_occurences.py [hunk 5]
```diff
@@ -0,0 +1,10 @@
+user_string = input("Text:").split()
+strings_dict = {}
+
+# for string in user_string:
+#     if string in strings_dict:
+#         strings_dict[string] += 1
+#     else:
+#         strings_dict[string] = 1
```
Comment (j-finger) [inline Prac 5/word_occurences.py]: This is functional code, you should use it :)

File: Prac 5/word_occurences.py [hunk 6]
```diff
@@ -0,0 +1,10 @@
...
+#     string_frequency = strings_dict[string]
+#     print("{} : {}".format(string, string_frequency))
```
Comment (j-finger) [inline Prac 5/word_occurences.py]: Is _string_ valid outside of the for loop on line 4? Perhaps a **_list_** is **_key_** to printing all the words in the dictionary?

**Review by j-finger — CHANGES_REQUESTED** [review]
I suggest watching the lecture recordings very closely and playing along in the python terminal and with a test file. Just try to do one week at a time (chronological order is almost essential) and you'll get the hang of it.

**Comment by lindsaymarkward** [comment]
@j-finger - this is one of the best code reviews I've seen in CP1404. Thanks for doing a good job 👍

**Pull request merged**